macros/models/_sector/dex/dex_atomic_arbitrages.sql

```diff
@@ -0,0 +1,129 @@
+{% macro dex_atomic_arbitrages(blockchain, transactions) %}
+
+-- Step 1: Check that the transaction contains at least 2 trades
+WITH multi_trade_txs AS (
+    SELECT block_time, tx_hash
+    FROM {{ ref('dex_trades') }}
```

Comment:
we're reading from `dex.trades` multiple times throughout. i wonder if performance wise, it's better to have a simple CTE at the top where you read from `dex.trades`, filter on blockchain and incremental reads, only pull columns which you need downstream. then each subsequent CTE reads from that one and does each step along the way.

Reply:
do you mind taking note of current CI test runtimes with code as-is now, then push a commit to try that, and see if we see any significant gains?

Reply:
based on monitored run times, the previous setup was most efficient. plz revert back and we can continue 🙏